go/vt/vtgate/executor_select_test.go

```diff
@@ -113,6 +114,96 @@ func TestSelectDBA(t *testing.T) {
 	utils.MustMatch(t, wantQueries, sbc1.Queries)
 }
 
+func TestSystemVariablesUsingSetVarComment(t *testing.T) {
+	executor, sbc1, _, _ := createExecutorEnv()
+	executor.normalize = true
+
+	session := NewSafeSession(&vtgatepb.Session{TargetString: "TestExecutor", SystemVariables: map[string]string{"sql_mode": "only_full_group_by"}})
```

Comment:
we should not set the session system variables directly then it would not test if there is any bug in calling set <sys_var> query.

instead, we should call executor.Execute with the set statement and then check the session for the InReservedConn value.